### DISCUSSION TITLE
Custom Ruleset error notifications.

### DISCUSSION THREAD

**Discussion by Coppertine** [ppy/osu#18510] [Ideas] [answered]
To explain, many custom rulesets are extremely in development and things can break.
An issue might happen for a user about a custom ruleset, but they would get a generic notification (possibly a link to the log files and a error report to Sentry), this might lead users to use the osu! github issue page for them to find out they are supposed to contact said ruleset developer.

I would propose the ability to either have the ruleset itself (Non Legacy) to get a link to the github issue page or to tell the user something like `An unknown error has occured with ruleset: "tau", please contact the ruleset developer.` So that custom ruleset issues are sorted and sent to the correct people instead of going to the lazer github issues page.
Another idea I have is to allow the ruleset itself to throw error codes so that debugging spesific issues are extremely easier for the developer's side. i.e `An unknown error has occured with ruleset: "tau", please contact the ruleset developer and provide the code "Editor-Selection"` or something similar.

**Comment by peppy** [comment, marked as answer]
https://github.com/ppy/osu/issues/9348
